### ISSUE TITLE
EOF:  IDENTIFICATION DIVISION.

### ISSUE THREAD

**Issue by GitMensch** [mridoni/gixsql#39]
Interestingly I've copied the sample lob source 1:1 via the browser, then saved and tried to preprocess it:

~~~
$ gixpp - the ESQL preprocessor for Gix-IDE/GixSQL
Version: 1.0.14
libgixpp version: 1.0.14

$ gixsql ./TEST.pco ./TEST.cob
EOF:   IDENTIFICATION DIVISION.

$ echo $?
255
~~~

Any idea what is happening there? Other files preprocessed with this version work fine, failing file attached,

_Originally posted by @GitMensch in https://github.com/mridoni/gixsql/issues/3#issuecomment-1113266100_

[TEST.pco.txt](https://github.com/mridoni/gixsql/files/8593061/TEST.pco.txt)

**Comment by dev4inno** [comment]
> Interestingly I've copied the sample lob source 1:1 via the browser, then saved and tried to preprocess it:
> 
> ```
> $ gixpp - the ESQL preprocessor for Gix-IDE/GixSQL
> Version: 1.0.14
> libgixpp version: 1.0.14
> 
> $ gixsql ./TEST.pco ./TEST.cob
> EOF:   IDENTIFICATION DIVISION.
> 
> $ echo $?
> 255
> ```
> 
> Any idea what is happening there? Other files preprocessed with this version work fine, failing file attached,
> 
> _Originally posted by @GitMensch in [#3 (comment)](https://github.com/mridoni/gixsql/issues/3#issuecomment-1113266100)_
> 
> [TEST.pco.txt](https://github.com/mridoni/gixsql/files/8593061/TEST.pco.txt)

I guess your input file has an incorrect format (IDENTIFICATION DIVISION starts at column 4).
Shifted every lines 4 bytes right and it works like a charm:

```
[haegar@linbookpro Release]$ ./gixpp -V
gixpp - the ESQL preprocessor for Gix-IDE/GixSQL
Version: 1.0.14
libgixpp version: 1.0.14

[haegar@linbookpro Release]$ ./gixpp -e -i TEST.pco -o TEST.cob -I .
;
;
;
[haegar@linbookpro Release]$ 
```
[TEST.pco.txt](https://github.com/mridoni/gixsql/files/8596698/TEST.pco.txt)
[TEST.cob.txt](https://github.com/mridoni/gixsql/files/8596699/TEST.cob.txt)

**Comment by GitMensch** [comment]
Of course! So let's put this issue to be about a nicer error message in this case.

**Comment by mridoni** [comment]
The problem is probably that, while pasting, the markdown formatter here on GitHub “eats” a few spaces, so when you copy and paste again it comes out in the wrong format

**Comment by mridoni** [comment]
I didn't actually close it, does by any chance GitHub auto-close issues when you write "Fixes#XXX" ?!

**Comment by GitMensch** [comment]
Yes, and I find that an important feature. Do you consider something "open" here? Fixed != fix released.

**Comment by mridoni** [comment]
> Yes, and I find that an important feature. Do you consider something "open" here? Fixed != fix released.

Well, one more thing to learn :smile:... I just didn't want to close it before it had some more testing, but it's probably 99.9% there.

**Comment by GitMensch** [comment]
As a new feature request (feel free to click on "Reference in new issue"):

You may want to look into adding a third (default) option to the reference-format: "auto". If no explicit reference-format is given gixpp would then do a "test-parsing", either by inspecting the first lines looking for directives/inline comments in col1-7, looking for * / D in column 7, or by running the parser "silently" first, if it fails try with free-form again and in each case re-running it in the final reference-format again (the manual pre-inspection looks more reasonable).

**Comment by mridoni** [comment]
Fixed in [v1.0.15](https://github.com/mridoni/gixsql/releases/tag/v1.0.15)
